app/gen-server/lib/HomeDBManager.ts

```diff
@@ -4745,16 +4732,53 @@ export class HomeDBManager extends EventEmitter {
     if (thisUser) { users.push(thisUser); }
     return { personal: true, public: !realAccess };
   }
+
+  private _getWorkspaceWithACLRules(scope: Scope, wsId: number) {
+    const query = this._workspace(scope, wsId, {
+      markPermissions: Permissions.VIEW
+    })
+    // Join the workspace's ACL rules (with 1st level groups/users listed).
+    .leftJoinAndSelect('workspaces.aclRules', 'acl_rules')
+    .leftJoinAndSelect('acl_rules.group', 'workspace_groups')
+    .leftJoinAndSelect('workspace_groups.memberUsers', 'workspace_group_users')
+    .leftJoinAndSelect('workspace_groups.memberGroups', 'workspace_group_groups')
+    .leftJoinAndSelect('workspace_group_users.logins', 'workspace_user_logins')
+    .leftJoinAndSelect('workspaces.org', 'org');
+    return verifyEntity(query);
+  }
+
+  private _buildOrgWithACLRulesQuery(scope: Scope, org: number|string, opts: Partial<QueryOptions> = {}) {
+    return this.org(scope, org, {
+      needRealOrg: true,
+      ...opts
+    })
+      // Join the org's ACL rules (with 1st level groups/users listed).
+      .leftJoinAndSelect('orgs.aclRules', 'acl_rules')
+      .leftJoinAndSelect('acl_rules.group', 'org_groups')
+      .leftJoinAndSelect('org_groups.memberUsers', 'org_member_users')
+      .leftJoinAndSelect('org_member_users.logins', 'user_logins');
+  }
+
+  private _getOrgWithACLRules(scope: Scope, org: number|string) {
+    const orgQuery = this._buildOrgWithACLRulesQuery(scope, org, {
+      markPermissions: Permissions.VIEW,
+      allowSpecialPermit: true,
+    });
+    return verifyEntity(orgQuery);
+  }
+
 }
 
 // Return a QueryResult reflecting the output of a query builder.
 // Checks on the "is_permitted" field which select queries set on resources to
 // indicate whether the user has access.
 // If the output is empty, we signal that the desired resource does not exist.
+// If we retrieve more than 1 entity, we signal that the request is ambiguous.
 // If the "is_permitted" field is falsy, we signal that the resource is forbidden.
 // Returns the resource fetched by the queryBuilder.
-async function verifyIsPermitted(
-  queryBuilder: SelectQueryBuilder<any>
+async function verifyEntity(
+  queryBuilder: SelectQueryBuilder<any>,
+  options: { discardPermissionsCheck?: boolean } = {}
```

Comment:
I think `skipPermissionCheck` might be a clearer name. Could you also update the function comment to describe its effect?

Reply:
Done!